shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/JDBCDatabaseCommunicationEngine.java

```diff
@@ -129,12 +135,33 @@ public ResponseHeader execute() throws SQLException {
             return new UpdateResponseHeader(executionContext.getSqlStatementContext().getSqlStatement());
         }
         proxySQLExecutor.checkExecutePrerequisites(executionContext);
-        List result = proxySQLExecutor.execute(executionContext);
-        refreshMetaData(executionContext);
-        Object executeResultSample = result.iterator().next();
-        return executeResultSample instanceof QueryResult
-                ? processExecuteQuery(executionContext, result, (QueryResult) executeResultSample)
-                : processExecuteUpdate(executionContext, result);
+        TransactionStatus transactionStatus = backendConnection.getConnectionSession().getTransactionStatus();
+        SQLStatement sqlStatement = executionContext.getSqlStatementContext().getSqlStatement();
+        JDBCBackendTransactionManager transactionManager = null;
+        if (TransactionType.XA == transactionStatus.getTransactionType() && transactionStatus.isInTransaction()
```

Comment:
Can you exclude the scenario of routing to a single data node? At this time, using the local transaction can meet the requirements.

Reply:
> Can you exclude the scenario of routing to a single data node? At this time, using the local transaction can meet the requirements.

@strongduanmu Sorry, I forgot to add the condition, I have changed it.